diff --git a/nixpkgs_review/github.py b/nixpkgs_review/github.py
index b62c72b622..edc0a660e1 100644
--- a/nixpkgs_review/github.py
+++ b/nixpkgs_review/github.py
@@ -1,4 +1,5 @@
 import json
+import os
 import shutil
 import tempfile
 import urllib.error
@@ -114,6 +115,12 @@
 
     def pull_request(self, number: int) -> Any:
         "Get a pull request"
+
+        if pr_json := os.environ.get("NIXPKGS_REVIEW_PULL_REQUEST_JSON"):
+            pr = json.loads(pr_json)
+            assert pr["number"] == number
+            return pr
+
         return self.get(f"repos/NixOS/nixpkgs/pulls/{number}")
 
     def get_json_from_artifact(self, workflow_id: int, json_filename: str) -> Any:
